release: v1.38.0 (#4494)
apollographql/router: .changesets/fix_lb_requires_selection_set_abstract_types.md

> ### fix: build entity representations with inline fragments correctly ([PR #4441](https://github.com/apollographql/router/pull/4441))
> 
> This uses a more flexible abstract type / concrete type check when applying a selection set to an entity reference before it's used in a fetch node. Previous to this change, we would drop data from the reference when it selected using an inline fragment (e.g. `@requires(fields: "... on Foo { a } ... on Bar { b }")`).

Reviewer: ```suggestion
### Fix building entity representations with inline fragments ([PR #4441](https://github.com/apollographql/router/pull/4441))

Previously, when applying a selection set to an entity reference before it's used in a fetch node, the router would drop data from the reference when it selected using an inline fragment, for example `@requires(fields: "... on Foo { a } ... on Bar { b }")`).

This release uses a more flexible abstract type / concrete type check when applying a selection set to an entity reference before it's used in a fetch node. 
```
